[CI] ShapeGeometryFieldMapperTests testCartesianBoundsBlockLoader failing

**Build Scans:**
- [elasticsearch-intake #20316 / part1](https://gradle-enterprise.elastic.co/s/3kr5ucdawphme)
- [elasticsearch-periodic-platform-support #6983 / ubuntu-2004_platform-support-unix](https://gradle-enterprise.elastic.co/s/xyr3swklkefeu)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.mapper.ShapeGeometryFieldMapperTests.testCartesianBoundsBlockLoader" -Dtests.seed=7684500F49392A5D -Dtests.locale=xh-ZA -Dtests.timezone=Africa/Blantyre -Druntime.java=24
```

**Applicable branches:**
8.18

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.mapper.ShapeGeometryFieldMapperTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testCartesianBoundsBlockLoader),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: geometry[0] 'BBOX (1.4987720674024342E-18, 1210127.625, -4.881798120682227E-36, -3.402822246271586E35)' wasn't extracted correctly
Expected: "    BBOX (1.4987720674024342E-18, 1210127.625, -4.881798120682227E-36, -3.402822246271586E35)"
     but: was <BBOX (-3.4028234663852886E38, 3.4028234663852886E38, 3.4028234663852886E38, -3.4028234663852886E38)>
```

**Issue Reasons:**
- [8.18] 2 failures in test testCartesianBoundsBlockLoader (0.3% fail rate in 685 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -332,9 +332,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/119396
 - class: org.elasticsearch.xpack.security.authc.ldap.ADLdapUserSearchSessionFactoryTests
   issue: https://github.com/elastic/elasticsearch/issues/119882
-- class: org.elasticsearch.index.mapper.AbstractShapeGeometryFieldMapperTests
-  method: testCartesianBoundsBlockLoader
-  issue: https://github.com/elastic/elasticsearch/issues/119201
 - class: org.elasticsearch.xpack.security.operator.OperatorPrivilegesIT
   method: testEveryActionIsEitherOperatorOnlyOrNonOperator
   issue: https://github.com/elastic/elasticsearch/issues/119911
@@ -450,9 +447,6 @@ tests:
 - class: org.elasticsearch.oldrepos.OldRepositoryAccessIT
   method: testOldRepoAccess
   issue: https://github.com/elastic/elasticsearch/issues/120148
-- class: org.elasticsearch.index.mapper.ShapeGeometryFieldMapperTests
-  method: testCartesianBoundsBlockLoader
-  issue: https://github.com/elastic/elasticsearch/issues/125129
 - class: org.elasticsearch.xpack.ilm.history.ILMHistoryItemTests
   method: testTruncateLongError
   issue: https://github.com/elastic/elasticsearch/issues/125216

--- a/server/src/test/java/org/elasticsearch/index/mapper/AbstractShapeGeometryFieldMapperTests.java
+++ b/server/src/test/java/org/elasticsearch/index/mapper/AbstractShapeGeometryFieldMapperTests.java
@@ -12,9 +12,10 @@ package org.elasticsearch.index.mapper;
 import org.apache.lucene.document.Document;
 import org.apache.lucene.geo.GeoEncodingUtils;
 import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
 import org.apache.lucene.index.LeafReader;
 import org.apache.lucene.store.Directory;
-import org.apache.lucene.tests.index.RandomIndexWriter;
 import org.elasticsearch.common.geo.GeometryNormalizer;
 import org.elasticsearch.core.Strings;
 import org.elasticsearch.geo.GeometryTestUtils;
@@ -39,14 +40,14 @@ import java.util.stream.IntStream;
 import static org.apache.lucene.geo.GeoEncodingUtils.decodeLongitude;
 import static org.elasticsearch.common.geo.Orientation.RIGHT;
 
-public class ShapeGeometryFieldMapperTests extends ESTestCase {
+public class AbstractShapeGeometryFieldMapperTests extends ESTestCase {
     public void testCartesianBoundsBlockLoader() throws IOException {
         testBoundsBlockLoader(
             CoordinateEncoder.CARTESIAN,
             () -> ShapeTestUtils.randomGeometryWithoutCircle(0, false),
             CartesianShapeIndexer::new,
             SpatialEnvelopeVisitor::visitCartesian,
-            ShapeGeometryFieldMapperTests::makeCartesianRectangle
+            AbstractShapeGeometryFieldMapperTests::makeCartesianRectangle
         );
     }
 
@@ -58,7 +59,7 @@ public class ShapeGeometryFieldMapperTests extends ESTestCase {
             () -> normalize(GeometryTestUtils.randomGeometryWithoutCircle(0, false)),
             field -> new GeoShapeIndexer(RIGHT, field),
             g -> SpatialEnvelopeVisitor.visitGeo(g, SpatialEnvelopeVisitor.WrapLongitude.WRAP),
-            ShapeGeometryFieldMapperTests::makeGeoRectangle
+            AbstractShapeGeometryFieldMapperTests::makeGeoRectangle
         );
     }
 
@@ -72,7 +73,7 @@ public class ShapeGeometryFieldMapperTests extends ESTestCase {
             geometries,
             field -> new GeoShapeIndexer(RIGHT, field),
             g -> SpatialEnvelopeVisitor.visitGeo(g, SpatialEnvelopeVisitor.WrapLongitude.WRAP),
-            ShapeGeometryFieldMapperTests::makeGeoRectangle
+            AbstractShapeGeometryFieldMapperTests::makeGeoRectangle
         );
     }
 
@@ -100,7 +101,9 @@ public class ShapeGeometryFieldMapperTests extends ESTestCase {
     ) throws IOException {
         var loader = new AbstractShapeGeometryFieldMapper.AbstractShapeGeometryFieldType.BoundsBlockLoader("field");
         try (Directory directory = newDirectory()) {
-            try (var iw = new RandomIndexWriter(random(), directory)) {
+            // Since we also test that the documents are loaded in the correct order, we need to write them in order, so we can't use
+            // RandomIndexWriter here.
+            try (var iw = new IndexWriter(directory, new IndexWriterConfig(null /* analyzer */))) {
                 for (Geometry geometry : geometries) {
                     var shape = new BinaryShapeDocValuesField("field", encoder);
                     shape.add(indexerFactory.apply("field").indexShape(geometry), geometry);

